server/src/main/java/io/crate/execution/dml/RawIndexer.java

```diff
@@ -92,26 +100,59 @@ public ParsedDocument index(IndexItem item) throws IOException {
                 returnValues
             );
         });
-        Object[] insertValues = new Object[doc.size()];
+
+        currentRowValues = new Object[doc.size()];
         Iterator<Object> iterator = doc.values().iterator();
-        List<Reference> columns = indexer.columns();
-        for (int i = 0; i < insertValues.length; i++) {
+        List<Reference> columns = currentRowIndexer.columns();
+        for (int i = 0; i < currentRowValues.length; i++) {
             Reference reference = columns.get(i);
             Object value = iterator.next();
             DataType<?> type = reference.valueType();
             try {
-                insertValues[i] = type.implicitCast(value);
+                currentRowValues[i] = type.implicitCast(value);
             } catch (ClassCastException | IllegalArgumentException e) {
                 throw new ConversionException(value, type);
             }
         }
-        ParsedDocument parsedDoc = indexer.index(new IndexItem.StaticItem(
+
```

Comment:
Similar to `Indexer`, `RawIndexer` also has index and this code is done separately.